android/library/src/main/java/io/logto/android/authflow/webview/AuthWebViewClient.kt

```diff
@@ -0,0 +1,47 @@
+package io.logto.android.authflow.webview
+
+import android.app.Activity
+import android.util.Log
+import android.webkit.WebResourceRequest
+import android.webkit.WebView
+import android.webkit.WebViewClient
+import io.logto.android.constant.AuthConstant
+import io.logto.android.utils.UrlUtil
+
+class AuthWebViewClient(
+    private val attachedActivity: Activity,
+    private val specifiedRedirectUri: String
+) : WebViewClient() {
+
+    override fun shouldOverrideUrlLoading(view: WebView?, request: WebResourceRequest?): Boolean {
+        if (view == null || request == null) {
+            return super.shouldOverrideUrlLoading(view, request)
```

Comment:
should we throw an error here?

Reply:
will be done in https://linear.app/silverhand/issue/LOG-67/catch-exceptions-in-webview-auth-flow